webapp/components/posts_view.jsx

```diff
@@ -425,6 +425,15 @@ export default class PostsView extends React.Component {
 
         return false;
     }
+    componentWillUpdate(nextProps) {
+        if (this.props.postList && this.props.postList.order.length) {
```

Comment:
Can you move this into componentWillReceiveProps and move showUnreadMessageAlert into the state? Since it's UI state, it should be part of that and not just another field on the component